src/components/util/equipmentInfosHandler.js

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { PARAM_USE_NAME } from '../../utils/config-params';
+
+export const useNameOrId = () => {
+    const useName = useSelector((state) => state[PARAM_USE_NAME]);
+    const getNameOrId = useCallback(
+        (name, id) => {
```

Comment:
Why not taking only an `equipmentObject` as parameter and use `equipmentObject?.id`, `equipmentObject?.name` inside this function ?

Reply:
because in some cases we don't have the same parameter name. 
For example : `equipment?.substationName,  equipment?.substationId`

Reply:
seems ok in your last revision